mln/models/module.py

```diff
@@ -59,14 +60,96 @@ def _calc_yield_info(self):
 		final_yield = min(time_yield + click_yield, yield_info.max_yield)
 		return final_yield, time_remainder, click_remainder
 
-	def calc_yield_qty(self):
-		"""Calculate the yield of this module."""
-		return self._calc_yield_info()[0]
+	def _distribute_items(self, clicker): 
+		"""Distribute items to owner, clicker, and owner's friends."""
+		for cost in self.item.execution_costs.all():
+			remove_inv_item(clicker, cost.item_id, cost.qty)
+		guest_yield = self._get_guest_yield(self.item.guest_yields.all())
+		if guest_yield is not None: 
+			add_inv_item(clicker, guest_yield.item_id, guest_yield.qty)
+		for owner_yield in self.item.owner_yields.all():
+			if random.random() < (owner_yield.probability / 100): 
+				add_inv_item(self.owner, owner_yield.item_id, owner_yield.qty)
+		for friend_message in self.item.friend_messages.all():
+			if random.random() < (friend_message.probability / 100): 
+				random_friend = self._get_random_friend()
+				send_template(template=friend_message.message, sender=self.owner, recipient=random_friend)

```

Comment:
Should be `break`s here

Reply:
Who is this addressed to? Is this resolved?